spec/unit/ssl/host_spec.rb

```diff
@@ -767,14 +766,63 @@ def base_json_comparison(result, json_hash)
 
       it "retrieves it from the server" do
         Puppet::Rest::Routes.expects(:get_crls)
-          .with(@http, Puppet::SSL::CA_NAME)
+          .with(@http, Puppet::SSL::CA_NAME, nil)
           .yields(@pki[:crl_chain])
 
         @host.ssl_store
         expect(Puppet::FileSystem.read(Puppet.settings[:hostcrl], :encoding => Encoding::UTF_8)).to eq(@pki[:crl_chain])
       end
     end
 
+    context "and the CRL is on disk" do
+      before do
+        @pki = PuppetSpec::SSL.create_chained_pki
+        localcacert = Puppet.settings[:localcacert]
+        hostcrl     = Puppet.settings[:hostcrl]
+        Puppet::Util.replace_file(localcacert, 0644) {|f| f.write @pki[:ca_bundle] }
+        Puppet::Util.replace_file(hostcrl, 0644)     {|f| f.write @pki[:crl_chain] }
+        @http = mock 'http'
+        @host.stubs(:http_client).returns(@http)
+      end
+
+      after do
+        Puppet::FileSystem.unlink(Puppet.settings[:localcacert])
+        Puppet::FileSystem.unlink(Puppet.settings[:hostcrl])
+      end
+
+      context "and there is a newer crl available from the master" do
+        it "retrieves the new crl from the server and replaces the crl on disk" do
+          pki2 = PuppetSpec::SSL.create_chained_pki
```

Comment:
The `create_chained_pki` is a kind of an expensive method, and I don't think we gain a whole lot here from running it again to get an actual `crl_chain`. Would it make sense to stub out some text for the yield block and just check that yield block?

Reply:
I could try.  Is there  a reason you used this method on line 755 instead of stubbing it out with some text? ( I was using that spec as a model).

Reply:
Ok.  Tried it with a simple string and seems to work.  Updated :)

Reply:
Yeah, I think when I wrote that test, I had some error checking in place to verify the contents of the file was a indeed a valid crl. The way the downloading works now, that doesn't happen.

I think additionally it would be good to refactor the tests to only call that method once; it's a read only sort of creation that would shave some time off the hosts test file. But that certainly doesn't have to happen in this PR.